src/main/java/org/zalando/nakadi/domain/AdminAuthorization.java

```diff
@@ -0,0 +1,89 @@
+package org.zalando.nakadi.domain;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.zalando.nakadi.plugin.api.authz.AuthorizationAttribute;
+import org.zalando.nakadi.plugin.api.authz.AuthorizationService;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class AdminAuthorization {
+
+    @NotNull
+    @Valid
+    @Size(min = 1, message = "must contain at least one attribute")
+    @JsonDeserialize(contentAs = AdminAuthorizationAttribute.class)
+    private final List<AuthorizationAttribute> admins;
+
+    @NotNull
+    @Valid
+    @Size(min = 1, message = "must contain at least one attribute")
+    @JsonDeserialize(contentAs = AdminAuthorizationAttribute.class)
+    private final List<AuthorizationAttribute> readers;
+
+    @NotNull
+    @Valid
+    @Size(min = 1, message = "must contain at least one attribute")
+    @JsonDeserialize(contentAs = AdminAuthorizationAttribute.class)
+    private final List<AuthorizationAttribute> writers;
+
+    public AdminAuthorization(@JsonProperty("admins") final List<AuthorizationAttribute> admins,
+                              @JsonProperty("readers") final List<AuthorizationAttribute> readers,
+                              @JsonProperty("writers") final List<AuthorizationAttribute> writers) {
+        // actually these three properties should never be null but the validation framework first creates an object
+        // and then uses getters to check if values are null or not, so we need to do this check to avoid exception
+        this.admins = admins == null ? null : Collections.unmodifiableList(admins);
+        this.readers = readers == null ? null : Collections.unmodifiableList(readers);
+        this.writers = writers == null ? null : Collections.unmodifiableList(writers);
+    }
+
+    public List<AuthorizationAttribute> getAdmins() {
+        return admins;
+    }
+
+    public List<AuthorizationAttribute> getReaders() {
+        return readers;
+    }
+
+    public List<AuthorizationAttribute> getWriters() {
+        return writers;
+    }
+
+    public List<AuthorizationAttribute> getList(final AuthorizationService.Operation operation) {
+        switch (operation) {
+            case ADMIN:
+                return admins;
+            case READ:
+                return readers;
+            case WRITE:
+                return writers;
+            default:
+                return null;
```

Comment:
and that's why it usually trhows an exception ) that later you don't dig into the stack trace understanding where this npe came from.